### ISSUE TITLE
URGENT: Get CIS Backdoor Working on MSK EKS Portal

### ISSUE THREAD

**Issue by averyniceday** [knowledgesystems/pipelines-scrum#707]
## Done Condition (What do we need? Why do we need it? Keep this is small as possible!)
This shoudl be done before we rollout to production -- need to make sure that the CIS backdoor works. Check with Avery or Ben for the wgets to use for testing. 

## Technical Description (How are we going to achieve the above)
We usually package in special stuff for the MSK Portal war -- this is slightly different because we're working in k8s. Make sure this still works.

**Comment by averyniceday** [comment]
application security XML is in the deployed image/container at the following path:
`/cbioportal-webapp/WEB-INF/lib/security-spring-0-unknown-version-SNAPSHOT.jar`

```
cp $PORTAL_CONFIG_HOME/properties/mskcc/*.properties $PORTAL_HOME/src/main/resources/
cp $PORTAL_CONFIG_HOME/properties/mskcc/applicationContext-security.xml $PORTAL_HOME/security/security-spring/src/main/resources/
cp $PORTAL_CONFIG_HOME/spring-security/MSKController.java $PORTAL_HOME/web/src/main/java/org/cbioportal/web
cp $PORTAL_CONFIG_HOME/spring-security/MSKCCRestfulAuthenticationSuccessHandler.java $PORTAL_HOME/security/security-spring/src/main/java/org/cbioportal/security/spring/authentication/saml/
cat $PORTAL_CONFIG_HOME/properties/mskcc/postLoadForMskCIS.js >> $PORTAL_HOME/portal/src/main/webapp/WEB-INF/jsp/global/header_bar.jsp
```

**Comment by averyniceday** [comment]
Step 1) We're changing the deployment file so webapp runner uses the unzipped cbioportal directory instead of the app.war

Step 2) Compile the MSK-specific files and whole cbioportal - happening inside docker 

Step 3) We're changing the deployment file to mount the MSK specific files and move them into the matching directories inside the cbioportal directory WEB-INF

**Comment by averyniceday** [comment]
Ben's option 
Step 1) Add java files to the portal codebase (with the conditional tag for controller) (what about .js file?) - rename to not include MSK in names

Step 2) Modify current application context security to use a "profile" specific for msk jlogin - Make username/password/authorities into secrets/properties

Step 3) Add properties to configmap for username/password/authorities/profile

Step 4) Update configmap with new props. 

** if we don't include the js file we might have to mount

**Comment by mandawilson** [comment]
The JS file is appended to this file `portal/src/main/webapp/WEB-INF/jsp/global/header_bar.jsp` and contains this:
```
<script>
function postLoadForMskCIS() {
try {
        var userDiv = document.getElementsByClassName('username');
        if (userDiv.length===1) {
                userDiv[0].remove();
        }
} catch(err) {
}
}
</script>
``` 

So we should just check a property or profile and remove the username if necessary.

- [ ] Check if the above code even works anymore.

**Comment by averyniceday** [comment]
note: {noop} might be deprecated

**Comment by mandawilson** [comment]
My stuff is in GitHub `origin/add-saml-plus-basic-auth`.

docker build -t "mandaewilson/cbioportal:saml-plus-basic-5" -f docker/web/Dockerfile . 
docker push mandaewilson/cbioportal:saml-plus-basic-5

Notes:
- applicationContext-security.xml has all old profiles (including saml) and now saml_plus_basic.  It might need the following added and uncommented in the saml_plus_basic, replacing the current SAMLContextProvider.
```
<!-- Provider of default SAML Context -->
    <!-- MEW: this is specific to this file too -->
    <!--b:bean id="contextProvider" class="org.springframework.security.saml.context.SAMLContextProviderLB">
      <b:property name="scheme" value="https"/>
      <b:property name="serverName" value="cbioportal.mskcc.org"/>
      <b:property name="serverPort" value="443"/>
      <b:property name="includeServerPortInRequestURL" value="true"/>
      <b:property name="contextPath" value="/"/>
    </b:bean-->
```

I couldn't get ping or saml loggin working with any docker image so I decided to leave that for later and only test the backdoor.  It does not seem to be working.  Watch the logs while you do the following:

wget --keep-session-cookies --save-cookies cookies.txt "http://eks-demo.cbioportal.org/j_spring_security_check?j_username=XXX&j_password=XXX"
wget --load-cookies cookies.txt "http://eks-demo.cbioportal.org/api/studies"
wget --load-cookies cookies.txt "http://eks-demo.cbioportal.org/j_spring_security_logout/"

The current error I am seeing in the log is about the JSP files so I am including the JS code from above to see if that fixes it.  I didn't check if the backdoor login actually worked.  I should make sure the user shows up in the logs.

Also I think the j_spring_security_logout is being used for front end users and doesn't work.  Test in the real MSK portal too.  I think this file needs modifying: `portal/reactapp/main.app.js` `{key:"logoutUrl",get:function(){return"saml"===this.authMethod?"saml/logout":"j_spring_security_logout"}}` and maybe the frontend?  I manually modified it.  It is ignored by git so it probably is in frontend code or something.

NOTE I modified the CONFIG_MAP in production and applied it and I started modifying the deployment file but did not apply it.
